test/marks/auto-test.js

```diff
@@ -10,7 +10,20 @@ it("Plot.autoSpec makes a histogram from a quantitative dimension", () => {
     y: {value: null, reduce: "count", zero: true},
     color: {value: null, reduce: null},
     size: {value: null, reduce: null},
-    mark: "bar"
+    mark: "bar",
+    markImpl: "rectY",
+    markOptions: {
+      fx: undefined,
+      fy: undefined,
+      x: {value: Object.assign([1, 1, 38], {label: "value"})},
+      y: undefined,
+      fill: undefined,
+      z: undefined,
+      r: undefined
+    },
```

Comment:
Remind me why we need to expose `markOptions`? And `transformOptions`?

Reply:
Yeah it does feel somewhat redundant with the other stuff we pass down. But it's the smallest change that makes this all work. 🤷 

markOptions and transformOptions are used to ultimately invoke the mark and transform implementations (and markOptions is the one place we pass down the materialized columns):

https://github.com/observablehq/plot/blob/e69d9b0b20427c56c58ab8ab0e9a7bd7d6e97e56/src/marks/auto.js#L295

If we wanna fold autoImpl back into autoSpec, we have to either expose markOptions and transformOptions from autoSpec, or down everything that goes into them. The advantage of having them in autoSpec is that it's easier to write a different "compiler" for the last mile, e.g. to construct a string instead of actually invoking the mark and transform: https://observablehq.com/@tophtucker/convert-plot-auto-to-explicit-marks.

It does sorta weird that we pass down y reduce info in both of these ways:

https://github.com/observablehq/plot/blob/e69d9b0b20427c56c58ab8ab0e9a7bd7d6e97e56/src/marks/auto.js#L142-L143

https://github.com/observablehq/plot/blob/e69d9b0b20427c56c58ab8ab0e9a7bd7d6e97e56/src/marks/auto.js#L186

But, the first is in the shape an explicit mark expects (composable transforms held apart from channels); the latter is in the shape the auto mark expects (reducers that imply transforms inside the channel options). If we wanna both 

- return a version of the auto mark options with all the defaults filled in (needed to invoke autoSpec idempotently in a call to auto, or to tell the notebook editor how an auto mark will behave), and
- support either immediately instantiating the mark or representing it as a string without repeating the options construction (needed for ejecting to explicit marks)
 
, then it feels reasonable to return with some redundancy in the shape.